tasks/main.yml

```diff
@@ -0,0 +1,89 @@
+---
+- name: Create consul group
+  group:
+    name: consul
+    system: yes
+
+- name: Create consul user
+  user:
+    name: consul
+    group: consul
+    createhome: no
+    system: yes
+
+- name: Mount "{{ consul_data_dir }}"
+  mount:
+    name: "{{ consul_data_dir }}"
+    src: "{{ consul_openstack_data_device }}"
+    fstype: ext4
+    state: mounted
+  when: consul_server
+
+- name: Create consul directories
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: consul
+    group: consul
+    mode: 0750
+  with_items:
+    - "{{ consul_config_dir }}"
+    - "{{ consul_data_dir }}"
+
+- name: Download and unarchive the Consul binary
+  unarchive:
+    src: "{{ consul_url }}"
+    dest: "{{ consul_bin_dir }}"
+    owner: consul
+    group: consul
+    remote_src: yes
+    creates: "{{ consul_bin_dir }}/consul"
+
+- name: Create Consul configuration file
+  template:
+    src: config.json.j2
+    dest: "{{ consul_config_dir }}/config.json"
+    owner: consul
+    group: consul
+
+- name: Create Consul service definition file
+  copy:
+    content: "{{ consul_service_config | to_nice_json(indent=4) }}"
+    dest: "{{ consul_config_dir }}/service.json"
+    owner: consul
+    group: consul
+  when: consul_service_config != ""
+
+- name: Create Consul systemd service file
+  template:
+    src: consul.service.j2
+    dest: /etc/systemd/system/consul.service
+
+- name: Enable and start systemd service to start on boot automatically
+  systemd:
+    name: consul.service
+    enabled: yes
+    state: started
+    daemon_reload: yes
+  notify: restart consul
```

Comment:
Why do we need to trigger the handler here?  I would expect this in a few of the other tasks, but not really here.  E.g. if we change the configuration file, Consul won't be restarted, which seems wrong.

Reply:
Yeah I realized this is probably unnecessary, but definitely necessary when we change the config. Moving it there :+1: